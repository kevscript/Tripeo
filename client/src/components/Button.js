import React from 'react'
import styled from 'styled-components'

const ButtonContainer = styled.button`
  border: 0;
  outline: 0;
  cursor: pointer;
  margin: ${props => props.big ? '20px 0' : '10px 0'};
  padding: ${props => props.big ? '15px 45px' : '10px 20px'};
  font-size: ${props => props.big ? '26px' : '15px'};
  border-radius: 50px;
  background: ${props => props.theme.colors.primary};
  color: ${props => props.theme.text.light};

  &:disabled {
    cursor: not-allowed;
    background: rgba(0,0,0,0.1);
  }
`

ButtonContainer.displayName = 'ButtonContainer'

const Button = ({ children, big, handleClick, disabled }) => {
  return (
    <ButtonContainer big={big} disabled={disabled} onClick={handleClick}>
      {children}
    </ButtonContainer>
  )
}

export default Button