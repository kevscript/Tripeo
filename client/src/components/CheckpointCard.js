import React from 'react'
import styled from 'styled-components'
import deleteBtn from '../assets/icons/delete-button.svg'
import { connect } from 'react-redux'
import { deleteCheckpoint } from '../actions'
import PropTypes from 'prop-types'

const CardContainer = styled.div`
  position: relative;
  display: flex;
  flex-direction: column;
  background: #f5f5f5;
  width: 100%;
  height: 80px;
  align-items: flex-start;
  justify-content: center;
  padding: 0 30px;
  margin: 2.5px 0;
`

const LocationName = styled.h3`
  font-size: 16px;
  font-weight: 600;
`

const LocationDate = styled.span`
  font-size: 16px;
`

const DeleteButton = styled.div`
  cursor: pointer;
  position: absolute;
  top: 15px;
  right: 15px;
  width: 15px;
  height: 15px;
`

const DeleteImg = styled.img`
  display: block;
  width: 100%;
  height: auto;
`

export const CheckpointCard = ({ checkpoint, deleteCheckpoint }) => {

  const { location, startDate, endDate } = checkpoint

  const handleDelete = () => {
    deleteCheckpoint(startDate)
  }

  return (
    <CardContainer data-test="CardContainer">
      <LocationName data-test="LocationName">
        {location.name}
      </LocationName>
      <LocationDate data-test="LocationDate">
        {startDate === endDate ? startDate : startDate + ' to ' + endDate}
      </LocationDate>
      <DeleteButton onClick={handleDelete} data-test="DeleteButton">
        <DeleteImg src={deleteBtn} alt="delete button" />
      </DeleteButton>
    </CardContainer>
  )
}

const mapDispatchToProps = {
  deleteCheckpoint
}

CheckpointCard.propTypes = {
  checkpoint: PropTypes.object,
  deleteCheckpoint: PropTypes.func,
}

export default connect(null, mapDispatchToProps)(CheckpointCard)