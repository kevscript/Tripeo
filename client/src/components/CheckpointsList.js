import React from 'react'
import styled from 'styled-components'
import LocationCard from './LocationCard'

const ListContainer = styled.ul`
  width: 100%;
  list-style: none;
  margin: 0;
`

const CheckpointsList = ({ checkpoints }) => {
  return (
    <ListContainer>
      {checkpoints.map((checkpoint, i) => {
        return <LocationCard key={i} checkpoint={checkpoint} />
      })}
    </ListContainer>
  )
}

export default CheckpointsList