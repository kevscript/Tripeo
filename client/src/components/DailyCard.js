import React from 'react'
import styled from 'styled-components'

const DailyItem = styled.li`
  display: flex;
  flex-direction: column;
  width: 100%;
  border: 1px solid rgba(0,0,0,0.5);
  padding: 30px;
`

const Row = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
`

const Text = styled.p`
`

const DailyCard = ({ forecast, cp }) => {
  return (
    <div>
      <DailyItem key={cp.timestamp}>
        <Row>
          <Text>{cp.location.name}</Text>
          <Text>{new Date(cp.timestamp).toLocaleDateString()}</Text>
        </Row>
        <Row>
          <Text>{forecast.daily.data[0].summary}</Text>
        </Row>
      </DailyItem>
    </div>
  )
}

export default DailyCard