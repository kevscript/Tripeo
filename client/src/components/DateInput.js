import React from 'react'
import DatePicker from 'react-datepicker'
import "react-datepicker/dist/react-datepicker.css"
import styled from 'styled-components'

const StyledDateInput = styled(DatePicker)`
  padding: 10px 15px;
  outline: 0;
  border: 1px solid lightblue;
  border-radius: 3px;
  font-size: 15px;
  margin: 10px 0;
`

const DateInput = ({ selectedDate, handleDateChange, placeholderText, minDate }) => {
  return (
    <StyledDateInput
      //showTimeSelect
      placeholderText={placeholderText}
      selected={selectedDate}
      onChange={handleDateChange}
      //timeFormat="HH:mm"
      //timeIntervals={60}
      //dateFormat="d MMMM yyyy"
      //timeCaption="time"
      minDate={minDate}
    />
  )
}

export default DateInput

