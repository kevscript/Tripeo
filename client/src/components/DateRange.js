import React from 'react'
import DatePicker from 'react-datepicker'
import "../styles/datepicker.css"
import styled from 'styled-components'
import PropTypes from 'prop-types'

const RangeContainer = styled.div`
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
`

const StyledDateInput = styled(DatePicker)`
  padding: 10px 15px;
  outline: 0;
  border: 1px solid lightgrey;
  border-radius: 3px;
  font-size: 14px;
  font-weight: 600;
  width: 125px;
  color: ${props => props.theme.colors.primary};
  text-align: center;

  ::-webkit-input-placeholder { /* Chrome/Opera/Safari */
  font-weight: 400;
  }
  ::-moz-placeholder { /* Firefox 19+ */
    font-weight: 500;
  }
  :-ms-input-placeholder { /* IE 10+ */
    font-weight: 500;
  }
  :-moz-placeholder { /* Firefox 18- */
    font-weight: 500;
  }
`

const Label = styled.span`
  font-size: 14px;
  margin-bottom: 5px;
`

const InputContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 10px 0;
`

const DateRange = ({
  handleStartDateChange,
  startDate,
  startMin,
  handleEndDateChange,
  endDate,
  endMin
}) => {
  return (
    <RangeContainer data-test="RangeContainer">
      <InputContainer>
        <Label>From (included)</Label>
        <StyledDateInput
          placeholderText="dd/mm/yyyy"
          selected={startDate}
          selectsStart
          startDate={startDate}
          endDate={endDate}
          onChange={handleStartDateChange}
          dateFormat="dd/MM/yyyy"
          minDate={startMin}
          data-test="startRange"
          popperPlacement="bottom"
          onFocus={(e) => e.target.readOnly = true}
        />
      </InputContainer>
      <InputContainer>
        <Label>To (included)</Label>
        <StyledDateInput
          placeholderText="dd/mm/yyyy"
          selected={endDate}
          selectsEnd
          startDate={startDate}
          endDate={endDate}
          onChange={handleEndDateChange}
          dateFormat="dd/MM/yyyy"
          minDate={endMin}
          data-test="endRange"
          popperPlacement="bottom"
          onFocus={(e) => e.target.readOnly = true}
        />
      </InputContainer>
    </RangeContainer>
  )
}

DateRange.propTypes = {
  handleStartDateChange: PropTypes.func.isRequired,
  handleEndDateChange: PropTypes.func.isRequired,
  startDate: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.instanceOf(Date)
  ]),
  endDate: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.instanceOf(Date)
  ]),
  startMin: PropTypes.instanceOf(Date),
  endMin: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.instanceOf(Date)
  ]),
}

export default DateRange
