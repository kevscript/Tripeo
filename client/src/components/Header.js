import React from 'react'
import styled from 'styled-components'
import { Link } from 'react-router-dom'

const HeaderContainer = styled.div`
  margin: 0 auto;
  width: 95%;
  max-width: 800px;
  height: 80px;
  display: flex;
  align-items: center;
  justify-content: space-between;
`

const HeaderNav = styled.ul`
  display: flex;
`

const NavItem = styled.li`
  margin: 0 10px;
`

const NavLink = styled(Link)`
  text-decoration: none;
  color: #333;
`

const HeaderLogo = styled.div`
  ${NavLink} {
    font-size: 36px;
    font-weight: 600;
  }
`

const Header = () => {
  return (
    <HeaderContainer>
      <HeaderLogo>
        <NavLink to='/'>TW</NavLink>
      </HeaderLogo>
      <HeaderNav>
        <NavItem>
          <NavLink to='/'>Your Trips</NavLink>
        </NavItem>
        <NavItem>
          <NavLink to='/'>Add Trip</NavLink>
        </NavItem>
      </HeaderNav>
    </HeaderContainer>
  )
}

export default Header