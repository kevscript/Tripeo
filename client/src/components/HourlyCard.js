import React from 'react'
import styled from 'styled-components'
import stampToTime from '../utils/stampToTime'
import Slider from 'react-slick'
import "slick-carousel/slick/slick.css";
import "slick-carousel/slick/slick-theme.css";
import "../styles/slider.css"

const CardContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  overflow-x: hidden;
`

const HourlyItem = styled.li`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  width: 100%;
  border-top: 1px solid rgba(0,0,0,0.2);
  border-bottom: 1px solid rgba(0,0,0,0.2);
  padding: 15px 15px;

  @media (min-width: 800px) {
    padding: 15px 30px;
    border-left: 1px solid rgba(0,0,0,0.2);
    border-right: 1px solid rgba(0,0,0,0.2);
  }
`

const TimeContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`

const StatsContainer = styled.div`
  min-height: 0px; 
  min-width: 0px;
  margin: 0 50px;
`

/*const StatsList = styled.ul`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  list-style: none;
`*/

const StatsItem = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100%;
`

const ItemName = styled.p`
  font-size: 10px;
  color: #333;
  text-align: center;
`

const ItemValue = styled.p`
  font-size: 14px;
  font-weight: 600;
  text-align: center;
`

const TempContainer = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;

  @media (min-width: 800px) {
    min-width: 100px;
  }
`

const IconContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 30px;
  height: 100%;
  margin: 0 10px;
`

const Icon = styled.img`
  display: block;
  width: 100%;
  height: auto;
`

const Text = styled.p `
font-weight: 600;
`

const HourlyCard = ({ data, timeZone }) => {
  const {
    time,
    temperature,
    humidity,
    cloudCover,
    windSpeed,
    uvIndex,
    visibility,
    icon
  } = data

  const sliderSettings = {
    arrow: true,
    speed: 800,
    slidesToShow: 4,
    slidesToScroll: 1,
    swipeToSlide: true,
    responsive: [
      {
        breakpoint: 800,
        settings: {
          slidesToShow: 2,
          slidesToScroll: 1,
        }
      },
      {
        breakpoint: 600,
        settings: {
          slidesToShow: 1,
          slidesToScroll: 1,
        }
      }
    ]
  }

  return (
    <CardContainer>
      <HourlyItem>
        <TimeContainer>
          <Text>{stampToTime(time, timeZone)}</Text>
        </TimeContainer>
        <StatsContainer>
          <Slider {...sliderSettings}>
            <StatsItem>
              <ItemName>Humidity (%)</ItemName>
              <ItemValue>{Math.round(humidity * 100)}</ItemValue>
            </StatsItem>
            <StatsItem>
              <ItemName>Wind (km/h)</ItemName>
              <ItemValue>{Math.round(windSpeed * 10) / 10}</ItemValue>
            </StatsItem>
            <StatsItem>
              <ItemName>UV Index</ItemName>
              <ItemValue>{uvIndex}</ItemValue>
            </StatsItem>
            <StatsItem>
              <ItemName>Visiblity (km)</ItemName>
              <ItemValue>{Math.round(visibility * 10) / 10}</ItemValue>
            </StatsItem>
            <StatsItem>
              <ItemName>Cloudiness (%)</ItemName>
              <ItemValue>{Math.round(cloudCover * 100)}</ItemValue>
            </StatsItem>
          </Slider>
        </StatsContainer>
        <TempContainer>
          <IconContainer>
            <Icon src={require(`../assets/icons/${icon}.svg`)} alt="weather icon" />
          </IconContainer>
          <Text>{Math.round(temperature)}Â°C</Text>
        </TempContainer>
      </HourlyItem>
    </CardContainer>
  )
}

export default HourlyCard