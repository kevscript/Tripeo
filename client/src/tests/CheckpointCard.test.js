import React from 'react'
import { CheckpointCard } from '../components/CheckpointCard'
import { shallow } from 'enzyme'

describe('CheckpointCard component', () => {

  let props = {
    checkpoint: {
      location: {
        name: 'Paris'
      },
      startDate: '01/02/2018',
      endDate: '01/02/2018'
    },
    deleteCheckpoint: jest.fn()
  }

  let component
  let wrapper

  beforeEach(() => {
    component = shallow(<CheckpointCard {...props} />)
    wrapper = component.find('[data-test="CardContainer"]')
  })

  test('should have a wrapper', () => {
    expect(wrapper.length).toBe(1)
  })

  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('should display the correct location name in LocationName element', () => {
    const element = wrapper.find('[data-test="LocationName"]')
    expect(element.text()).toBe('Paris')
  })

  test('should properly display and format the right date string', () => {
    const element = wrapper.find('[data-test="LocationDate"]')
    expect(element.text()).toBe('01/02/2018')
  })

  test('should have a button element', () => {
    const element = wrapper.find('[data-test="DeleteButton"]')
    expect(element.length).toBe(1)
  })

  test('should call the right click handler when clicking button', () => {
    const element = wrapper.find('[data-test="DeleteButton"]')
    element.simulate('click')
    expect(props.deleteCheckpoint).toHaveBeenCalled()
  })
})