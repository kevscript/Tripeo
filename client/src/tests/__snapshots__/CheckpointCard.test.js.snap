// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckpointCard component should render correctly 1`] = `ShallowWrapper {}`;
